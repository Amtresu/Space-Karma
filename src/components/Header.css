.header {
  display: flex;
  flex-direction: row;
  margin: 20px;
  justify-content: space-between;
  align-items: center;
}

img {
  width: 10%;
}

.header-photo {
  display: flex;
  gap: 20px;
}
