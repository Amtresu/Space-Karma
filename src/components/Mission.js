import React from 'react';
import { useDispatch } from 'react-redux';
import { toggleMission } from '../redux/missions/mission';
import './mission.css';

const Mission = (props) => {
  const { id, name, description, joined } = props;
  const dispatch = useDispatch();

  const toggleMissions = () => {
    dispatch(toggleMission(id));
  };

  const button = joined ? 'Joined' : 'Join Mission';
  const reserve = joined ? 'Reserved' : 'Reserve';
  const member = joined ? 'member' : 'non-member';

  return (
    <tr className="mission-list" key={id}>
      <td className="mission-title">{name}</td>
      <td className="mission-description">{description}</td>
      <td className="mission-status">
        <button className={member}>{reserve}</button>
      </td>
      <td className="mission-button">
        <button
          id={id}
          type="button"
          className={reserve}
          onClick={toggleMissions}
        >
          {button}
        </button>
      </td>
    </tr>
  );
};
export default Mission;
