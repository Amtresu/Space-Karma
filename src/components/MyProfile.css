.my-profile {
    display: flex;
    font-size: 25px;
}

.main{
    width: 50%;
    text-align: center;
}

tr {
    height: 100px;
}

