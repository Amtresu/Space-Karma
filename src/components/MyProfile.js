/* eslint-disable */

import { useSelector } from 'react-redux';
import { ListGroup } from 'react-bootstrap';

function Myprofile() {
  const reservationRocket = useSelector((state) =>
    state.rocket.filter((rocket) => rocket.reserved)
  );
  return (
    <div className="main">
      <div className="rockets">
        <h2>Reserved Rockets</h2>
        <ListGroup>
          {reservationRocket.map((rocket) => (
            <ListGroup.Item key={rocket.id}>
              <h4>{rocket.name}</h4>
            </ListGroup.Item>
          ))}
        </ListGroup>
      </div>
    </div>
  );
}

export default Myprofile;
