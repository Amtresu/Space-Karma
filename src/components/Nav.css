.navigation {
  display: flex;
  gap: 50px;
  color: black;
}

.style {
  color: blue;
  text-decoration: none;
  font-size: 24px;
}

.style:hover {
  color: rgb(0, 0, 0);
}
