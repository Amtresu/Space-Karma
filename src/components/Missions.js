import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getMissions } from '../redux/missions/mission';
import Mission from './Mission';

function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission);

  useEffect(() => {
    dispatch(getMissions());
  }, []);

  return (
    <table>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <Mission
            key={mission.id}
            id={mission.id}
            name={mission.name}
            description={mission.description}
            joined={mission.joined}
          />
        ))}
      </tbody>
    </table>
  );
}
export default Missions;
