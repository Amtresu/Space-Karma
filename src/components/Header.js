/* eslint-disable */
import planet from '../images/planet.png';
import './Header.css';
import React from 'react';
import Nav from './Nav';

function Header() {
  return (
    <div className="header">
      <div className="header-photo">
        <img alt="logo" src={planet} />
        <h1 className="header-text">Space Karma Page</h1>
      </div>
      <Nav />
    </div>
  );
}

export default Header;
