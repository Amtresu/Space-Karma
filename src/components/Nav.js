/* eslint-disable */

import React from 'react';
import './Nav.css';
import { NavLink } from 'react-router-dom';
import { notInitialized } from 'react-redux/es/utils/useSyncExternalStore';

function Nav() {
  return (
    <div className="navigation">
      <NavLink
        style={({ isActive }) => ({
          color: isActive ? '#0000FF' : '#545e6f',
        })}
        to="/"
        className="style"
      >
        Rockets
      </NavLink>
      <NavLink
        style={({ isActive }) => ({
          color: isActive ? '#0000FF' : '#545e6f',
        })}
        to="Mission"
        className="style"
      >
        Mission
      </NavLink>
      <NavLink
        style={({ isActive }) => ({
          color: isActive ? '#0000FF' : '#545e6f',
        })}
        to="MyProfile"
        className="style"
      >
        My Profile
      </NavLink>
    </div>
  );
}
export default Nav;
