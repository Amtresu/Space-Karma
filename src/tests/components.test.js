/* eslint-disable */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Mission from '../components/Mission';
import Missions from '../components/Missions';
import Nav from '../components/Nav';
import Header from '../components/Header';
import MyProfile from '../components/MyProfile';
import Rocket from '../components/Rocket';
import Rockets from '../components/Rockets';


describe('Render All Components', () => {
    test('render navbar', () => {

        const { container } = render(
            <MemoryRouter>
                <Nav />
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
          class="navigation"
    >
      <a
        aria-current="page"
        class="style active"
        href="/"
        style="color: rgb(0, 0, 255);"
      >
        Rockets
      </a>
      <a
        class="style"
        href="/Mission"
        style="color: rgb(84, 94, 111);"
      >
        Mission
      </a>
      <a
        class="style"
        href="/MyProfile"
        style="color: rgb(84, 94, 111);"
      >
        My Profile
      </a>
    </div>
  `);
    });

    test('render missions', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Missions />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <table>
           <thead>
             <tr>
               <th>
                 Mission
               </th>
               <th>
                 Description
               </th>
               <th>
                 Status
               </th>
             </tr>
           </thead>
           <tbody />
         </table>
    `);
    });

    test('render mission', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Mission />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(  
        `<tr
      class="mission-list"
    >
      <td
        class="mission-title"
      />
      <td
        class="mission-description"
      />
      <td
        class="mission-status"
      >
        <button
          class="non-member"
        >
          Reserve
        </button>
      </td>
      <td
        class="mission-button"
      >
        <button
          class="Reserve"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>`
     );
    });

    test('render Header', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Header />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <div
          class="header"
        >
          <div
            class="header-photo"
          >
            <img
              alt="logo"
              src="planet.png"
            />
            <h1
              class="header-text"
            >
              Space Karma Page
            </h1>
          </div>
          <div
            class="navigation"
          >
            <a
              aria-current="page"
              class="style active"
              href="/"
              style="color: rgb(0, 0, 255);"
            >
              Rockets
            </a>
            <a
              class="style"
              href="/Mission"
              style="color: rgb(84, 94, 111);"
            >
              Mission
            </a>
            <a
              class="style"
              href="/MyProfile"
              style="color: rgb(84, 94, 111);"
            >
              My Profile
            </a>
          </div>
        </div>
    `);
    });

    test('render MyProfile', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <MyProfile />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <div className='my-profile'>
        <table className="main">
          <thead>
          <tr>
            <th>My Rockets</th>
          </tr>
          </thead>
          <tbody className="rockets">
                <tr>
                  <td></td>
                  </tr>
          </tbody>
        </table>
  
        <table className="main">
          <thead>
          <tr>
            <th>My Missions</th>
          </tr>
          </thead>
          <tbody className="missions">
                <tr>
                  <td></td>
          </tbody>
        </table>
      </div>
    `);
    });


    test('render Rockets', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Rockets />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <div>
        <div className="rocket-main">
          <Rocket/>
      </div>
    `);
    });

    test('render Rocket', () => {
        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Rocket />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(
         `<div>
           <img
             alt="rocket"
           />
           <h2 />
           <div
             class="reservation"
           >
             <button />
           </div>
           <button
             type="button"
           >
             Reserve Rocket
           </button>
         </div>`
    );
    });
})