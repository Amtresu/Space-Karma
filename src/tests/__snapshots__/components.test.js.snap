// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render All Components render Header: 
        <div
          class="header"
        >
          <div
            class="header-photo"
          >
            <img
              alt="logo"
              src="planet.png"
            />
            <h1
              class="header-text"
            >
              Space Karma Page
            </h1>
          </div>
          <div
            class="navigation"
          >
            <a
              aria-current="page"
              class="style active"
              href="/"
              style="color: rgb(0, 0, 255);"
            >
              Rockets
            </a>
            <a
              class="style"
              href="/Mission"
              style="color: rgb(84, 94, 111);"
            >
              Mission
            </a>
            <a
              class="style"
              href="/MyProfile"
              style="color: rgb(84, 94, 111);"
            >
              My Profile
            </a>
          </div>
        </div>
     1`] = `
<div
  class="header"
>
  <div
    class="header-photo"
  >
    <img
      alt="logo"
      src="planet.png"
    />
    <h1
      class="header-text"
    >
      Space Karma Page
    </h1>
  </div>
  <div
    class="navigation"
  >
    <a
      aria-current="page"
      class="style active"
      href="/"
      style="color: rgb(0, 0, 255);"
    >
      Rockets
    </a>
    <a
      class="style"
      href="/Mission"
      style="color: rgb(84, 94, 111);"
    >
      Mission
    </a>
    <a
      class="style"
      href="/MyProfile"
      style="color: rgb(84, 94, 111);"
    >
      My Profile
    </a>
  </div>
</div>
`;

exports[`Render All Components render MyProfile: 
        <div className='my-profile'>
        <table className="main">
          <thead>
          <tr>
            <th>My Rockets</th>
          </tr>
          </thead>
          <tbody className="rockets">
                <tr>
                  <td></td>
                  </tr>
          </tbody>
        </table>
  
        <table className="main">
          <thead>
          <tr>
            <th>My Missions</th>
          </tr>
          </thead>
          <tbody className="missions">
                <tr>
                  <td></td>
          </tbody>
        </table>
      </div>
     1`] = `
<div
  class="my-profile"
>
  <table
    class="main"
  >
    <thead>
      <tr>
        <th>
          My Rockets
        </th>
      </tr>
    </thead>
    <tbody
      class="rockets"
    />
  </table>
  <table
    class="main"
  >
    <thead>
      <tr>
        <th>
          My Missions
        </th>
      </tr>
    </thead>
    <tbody
      class="missions"
    />
  </table>
</div>
`;

exports[`Render All Components render Rocket: <div>
           <img
             alt="rocket"
           />
           <h2 />
           <div
             class="reservation"
           >
             <button />
           </div>
           <button
             type="button"
           >
             Reserve Rocket
           </button>
         </div> 1`] = `
<div>
  <img
    alt="rocket"
  />
  <h2 />
  <div
    class="reservation"
  >
    <button />
  </div>
  <button
    type="button"
  >
    Reserve Rocket
  </button>
</div>
`;

exports[`Render All Components render Rockets: 
        <div>
        <div className="rocket-main">
          <Rocket/>
      </div>
     1`] = `
<div
  class="rocket-main"
/>
`;

exports[`Render All Components render mission: <tr
      class="mission-list"
    >
      <td
        class="mission-title"
      />
      <td
        class="mission-description"
      />
      <td
        class="mission-status"
      >
        <button
          class="non-member"
        >
          Reserve
        </button>
      </td>
      <td
        class="mission-button"
      >
        <button
          class="Reserve"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr> 1`] = `
<tr
  class="mission-list"
>
  <td
    class="mission-title"
  />
  <td
    class="mission-description"
  />
  <td
    class="mission-status"
  >
    <button
      class="non-member"
    >
      Reserve
    </button>
  </td>
  <td
    class="mission-button"
  >
    <button
      class="Reserve"
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;

exports[`Render All Components render missions: 
        <table>
           <thead>
             <tr>
               <th>
                 Mission
               </th>
               <th>
                 Description
               </th>
               <th>
                 Status
               </th>
             </tr>
           </thead>
           <tbody />
         </table>
     1`] = `
<table>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Render All Components render navbar: 
          class="navigation"
    >
      <a
        aria-current="page"
        class="style active"
        href="/"
        style="color: rgb(0, 0, 255);"
      >
        Rockets
      </a>
      <a
        class="style"
        href="/Mission"
        style="color: rgb(84, 94, 111);"
      >
        Mission
      </a>
      <a
        class="style"
        href="/MyProfile"
        style="color: rgb(84, 94, 111);"
      >
        My Profile
      </a>
    </div>
   1`] = `
<div
  class="navigation"
>
  <a
    aria-current="page"
    class="style active"
    href="/"
    style="color: rgb(0, 0, 255);"
  >
    Rockets
  </a>
  <a
    class="style"
    href="/Mission"
    style="color: rgb(84, 94, 111);"
  >
    Mission
  </a>
  <a
    class="style"
    href="/MyProfile"
    style="color: rgb(84, 94, 111);"
  >
    My Profile
  </a>
</div>
`;
